go/libkb/merkle_client.go

```diff
@@ -774,10 +785,13 @@ func readSkipSequenceFromStringList(v []string) (ret SkipSequence, err error) {
 }
 
 func readRootFromAPIRes(m MetaContext, jw *jsonw.Wrapper, opts MerkleOpts) (*MerkleRoot, error) {
-	ret, err := NewMerkleRootFromJSON(m, jw, opts)
+	ret, err := NewMerkleRootFromJSON(jw, opts)
 	if err != nil {
 		return nil, err
 	}
+	if chk := GetMerkleCheckpoint(m); chk != nil && *ret.Seqno() < *chk.Seqno() {
+		return nil, NewClientMerkleFailedCheckpointError(fmt.Sprintf("got unexpected early root %d < %d", *ret.Seqno(), *chk.Seqno()))
```

Comment:
Assuming this is for noticing a server bug. This should log immediately so there's no chance the error is missed by the error value being dropped.